eds_scikit/phenotype/base.py

```diff
@@ -0,0 +1,421 @@
+import inspect
+import re
+from pathlib import Path
+from typing import Optional, Tuple
+
+from loguru import logger
+
+from eds_scikit.event import conditions_from_icd10, procedures_from_ccam
+from eds_scikit.io import BaseData
+from eds_scikit.utils.framework import bd
+from eds_scikit.utils.typing import DataFrame
+
+
+class Features:
+    """
+    Class used to store features (i.e. DataFrames). Features are
+    stored in the self._features dictionary.
+    """
+
+    REQUIRED_COLUMNS: set = {"person_id", "phenotype"}
+
+    def __init__(self):
+        self._features = {}
+        self.last_feature = None
+
+    def __bool__(self):
+        return bool(self._features)
+
+    def __repr__(self):
+        return "Features: " + ", ".join(self._features.keys())
+
+    def __contains__(self, feature_name: str):
+        return feature_name in self._features
+
+    def __getitem__(self, feature_name: str):
+        if feature_name not in self._features:
+            raise KeyError(f"No feature named {feature_name}")
+        return self._features[feature_name]
+
+    def __setitem__(self, feature_name: str, feature: DataFrame):
+        missing_columns = self.REQUIRED_COLUMNS - set(feature.columns)
+        if missing_columns:
+            raise ValueError(
+                f"Cannot add feature {feature_name}: Columns {missing_columns} are missing"
+            )
+
+        self._features[feature_name] = feature
+        self.last_feature = feature_name
+
+    def last(self):
```

Comment:
We should at least document this since this is a very weird behaviour in general